src/onboarding/Deployment/DeploymentStepsItem.js

```diff
@@ -89,6 +150,111 @@ DeploymentStepsItem.propTypes = {
   index: PropTypes.number.isRequired,
   name: PropTypes.string.isRequired,
   status: TransactionStatusType.isRequired,
+  dateStart: PropTypes.objectOf(Date),
+  gasPrice: PropTypes.number,
+}
+
+const StatusMessage = ({ status, remainingTime, remainingTimeUnit }) => {
+  switch (status) {
+    case TRANSACTION_STATUS_UPCOMING:
+      return ''
+    case TRANSACTION_STATUS_SIGNING:
+      return 'Waiting for signature'
+    case TRANSACTION_STATUS_PENDING:
+      const displayTime = ` - ${remainingTime} ${remainingTimeUnit}`
+      return `Transaction in progress${displayTime}`
+    case TRANSACTION_STATUS_SUCCESS:
+      return ''
+  }
+}
+
+StatusMessage.propTypes = {
+  status: TransactionStatusType.isRequired,
+  remainingTime: PropTypes.number,
+  remainingTimeUnit: PropTypes.string,
+}
+
+const ProgressRing = ({ value, diameter }) => {
+  const theme = useTheme()
+  return (
+    <RingContainer diameter={diameter}>
+      <RingBorder diameter={diameter} theme={theme} />
+      <Half side="right" value={value} diameter={diameter} />
+      <Half side="left" value={value} diameter={diameter} />
+    </RingContainer>
+  )
+}
+
+ProgressRing.propTypes = {
+  value: PropTypes.number.isRequired,
+  diameter: PropTypes.number.isRequired,
+}
+
+const Half = ({ side, value, diameter }) => {
+  const theme = useTheme()
+  let angle
+  if (side === 'right') {
+    if (value <= 0.5) {
+      angle = value * 360
+    } else {
+      angle = 180
+    }
+  } else {
+    if (value > 0.5) {
+      angle = value * 360
+    } else {
+      angle = 180
+    }
+  }
+  return (
+    <HalfHider side={side} diameter={diameter}>
+      <HalfRing angle={angle} diameter={diameter} theme={theme} />
+    </HalfHider>
+  )
+}
+
+Half.propTypes = {
+  side: PropTypes.oneOf(['right', 'left']),
+  value: PropTypes.number.isRequired,
+  diameter: PropTypes.number.isRequired,
 }
 
+const RingContainer = styled.div`
+  position: relative;
+  width: ${({ diameter }) => diameter}px;
+`
+
+const RingBorder = styled.div`
+  height: ${({ diameter }) => diameter}px;
+  border: 2px solid ${({ theme }) => theme.border};
+  border-radius: 50%;
+`
+
+const HalfHider = styled.div`
+  position: absolute;
+  top: 0px;
+  width: ${({ diameter }) => diameter}px;
+  height: ${({ diameter }) => diameter}px;
+  clip-path: inset(
+    ${({ side, diameter }) =>
+      side === 'right'
+        ? `0px 0px 0px ${diameter / 2}px`
+        : `0px ${diameter / 2}px 0px 0px`}
+  );
+`
+
+const HalfRing = styled.div.attrs(({ angle }) => ({
+  style: {
+    transform: `rotate(${angle}deg)`,
+  },
+}))`
+  position: absolute;
+  top: 0px;
+  width: ${({ diameter }) => diameter}px;
+  height: ${({ diameter }) => diameter}px;
+  border: 2px solid ${({ theme }) => theme.selected};
+  border-radius: 50%;
+  clip-path: inset(0px ${({ diameter }) => diameter / 2}px 0px 0px);
```

Comment:
It is considered a standard convention to remove the unit for zero values:
```suggestion
  clip-path: inset(0 ${({ diameter }) => diameter / 2}px 0 0);
```

Reply:
Same for lines 235, 241, 242, 252